### PR TITLE
Add PG::Connection#check_connection as a way to verify the socket is OK

### PR DESCRIPTION
**Pull request by larskanis**
... without data exchange with the server.

Related to #509

### CODE REVIEW

File: lib/pg/connection.rb [hunk 1]
```diff
@@ -343,6 +343,22 @@ def ssl_attributes
...
 
+	# Read all pending socket input to internal memory and raise an exception in case of errors.
+	#
+	# This verifies that the connection socket is in a usable state and not aborted in any way.
+	# No communication is done with the server.
+	# Only pending data is read from the socket - the method doesn't wait for any outstanding server answers.
+	#
+	# In order to check the communication with the server, it's recommended to send an empty query per #exec instead of using this method.
+	#
+	# Raises a kind of PG::Error if there was an error reading the data of if the socket is in a failure state.
```
Comment (mt-clearhaus) [inline lib/pg/connection.rb:354]:
```suggestion
	# Raises a kind of PG::Error if there was an error reading the data or if the socket is in a failure state.
```

**Review by ged — APPROVED** [review]
Looks good. The suggestion from @mt-clearhaus looks good too.

**Comment by larskanis** [comment]
I wonder if `check_connection` is the best name, as it sounds like a connection round trip is done. Maybe `check_socket` is better. What do you think?

**Comment by cbandy** [comment]
Agreed. I like `check_socket`.

**Comment by ged** [comment]
Yeah, `check_socket` is much better.

**Pull request merged**